File: backend/B2BApplication/Models/Catalogue.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;

namespace B2BApplication.Models
{
    public class Catalogue
    {
        public int CatalogueId { get; set; }
        public string CatalogueIntitule { get; set; }
        public int CatalogueNiveau { get; set; }
       
        public Catalogue? CatalogueParent { get; set; }
        [ForeignKey("CatalogueParent")]
        public int? CatalogueParentId  { get; set; }
    }
}

File: frontend/src/app/models/catalogue.ts
export class Catalogue{
catalogueId:number
catalogueIntitule:string
catalogueNiveau:number
catalogueParentId:number
catalogueParent:Catalogue|null=null
constructor(catalogueId:number,catalogueIntitule:string,catalogueNiveau:number,catalogueParentId:number){
this.catalogueId=catalogueId
this.catalogueIntitule=catalogueIntitule
this.catalogueNiveau=catalogueNiveau
this.catalogueParentId=catalogueParentId
}
}